search-app/src/cmr/search/services/query_execution/facets/granule_v2_facets.clj

```diff
@@ -63,18 +64,24 @@
 
 (defmethod v2-facets/create-v2-facets-by-concept-type :granule
   [concept-type base-url query-params aggs _]
-  (let [temporal-facets (temporal-facets/parse-temporal-buckets
-                         (-> aggs :start-date-doc-values :buckets)
-                         :year)
-        year-node (when (seq temporal-facets)
-                    (v2h/generate-group-node "Year"
-                                              false
-                                              temporal-facets))
-        ;; Will be added by CMR-4683 and CMR-4684
-        year-node (dissoc year-node :applied :type)
-        temporal-node (when year-node
-                        (v2h/generate-group-node (v2h/fields->human-readable-label :temporal)
-                                                 false
-                                                 [year-node]))
-        temporal-node (dissoc temporal-node :applied :type)]
-     [temporal-node]))
+  (let [subfacets (hv2/hierarchical-bucket-map->facets-v2
+                   :temporal-facet
+                   (:start-date-doc-values aggs)
+                   base-url
+                   query-params)]
+    (when (seq subfacets)
```

Comment:
I'm unsure if this should be a when-let or not, probably not but I'd thought I'd throw it out there just in case.

Reply:
No, this is good -- `when-let` only makes sense if you're assigning one value.